backend/src/packages/surveys/survey.controller.ts

```diff
@@ -0,0 +1,46 @@
+import { APIPath } from '#libs/enums/enums.js';
+import {
+  type APIHandlerOptions,
+  type APIHandlerResponse,
+  BaseController,
+} from '#libs/packages/controller/controller.js';
+import { HTTPCode } from '#libs/packages/http/http.js';
+import { type Logger } from '#libs/packages/logger/logger.js';
+import { AuthApiPath } from '#packages/auth/auth.js';
+
+import { type SurveyService } from './survey.service.js';
+import { type SurveyRequestDto } from './surveys.js';
+
+class SurveyController extends BaseController {
+  private surveyService: SurveyService;
+
+  public constructor(logger: Logger, surveyService: SurveyService) {
+    super(logger, APIPath.AUTH);
+
+    this.surveyService = surveyService;
+
+    this.addRoute({
+      path: AuthApiPath.SIGN_UP_SURVEY,
+      method: 'POST',
+      handler: (options) =>
+        this.create(
+          options as APIHandlerOptions<{
+            body: SurveyRequestDto;
+          }>,
+        ),
+    });
+  }
+
```

Comment:
We also need to add comments for swagger documentation like in user controller, you can check how it will look like on /documentation route

Reply:
@Liza-Veis Could, you please advise me, on how can I make this route(/documentation) to be public?
I'll do this, but it doesn't help me
![image](https://github.com/BinaryStudioAcademy/bsa-2023-calmpal/assets/102380564/6d577840-764d-40c9-93bc-6fffdf6fd07f)
![image](https://github.com/BinaryStudioAcademy/bsa-2023-calmpal/assets/102380564/010b7d27-26d4-45a0-85d1-aa0b9db44636)

Reply:
It's a good notice, we will create a bug ticket for this

Reply:
So, what I should do in this way? Should I wait for this to check how it will work?

Reply:
you can disable route protection for a while, check that your documentation works and then return protection back

Reply:
I commented initPlugins function in base-server-application. But I also can't get access to this route. Did it work before?
![image](https://github.com/BinaryStudioAcademy/bsa-2023-calmpal/assets/102380564/954a4daf-8456-410d-94ea-76b5a58db7f5)
![image](https://github.com/BinaryStudioAcademy/bsa-2023-calmpal/assets/102380564/b0933c97-998c-437e-8229-00370f9beeb5)

Reply:
I've noticed that `request.routerPath` (which is used for checking whether the path is a white route) in fastify hook sees the `api/v1/documentation` path as `/*`. To resolve this issue, we can change the `request.routerPath` to `request.url`. But is it a good option?

Reply:
I wrote a swagger documentation. but I can't check if it works correctly, because I can't get access to this route.
But I hope it does =)

Reply:
We have merged fix for swagger documentation in #79, you can try it with new changes, it should be accessible by /api/v1/documentation path